astroquery/mast/observations.py

```diff
@@ -0,0 +1,902 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+MAST Observations
+=================
+
+This module contains various methods for querying MAST observations.
+"""
+
+
+import warnings
+import json
+import time
+import os
+import uuid
+
+import numpy as np
+
+from requests import HTTPError
+
+import astropy.units as u
+import astropy.coordinates as coord
+
+from astropy.table import Table, Row, vstack, MaskedColumn
+from astropy.logger import log
+
+from astropy.utils import deprecated
+from astropy.utils.console import ProgressBarOrSpinner
+from astropy.utils.exceptions import AstropyDeprecationWarning
+
+from six.moves.urllib.parse import quote as urlencode
+
+from ..query import QueryWithLogin
+from ..utils import commons, async_to_sync
+from ..utils.class_or_instance import class_or_instance
+from ..exceptions import (TimeoutError, InvalidQueryError, RemoteServiceError,
+                          ResolverError, MaxResultsWarning,
+                          NoResultsWarning, InputWarning, AuthenticationWarning)
+
+from . import conf, utils, fpl
+from .auth import MastAuth
+from .cloud import CloudAccess
+from .discovery_portal import PortalAPI
+
+
+__all__ = ['Observations', 'ObservationsClass']
+
+
+@async_to_sync
+class ObservationsClass(QueryWithLogin):
+    """
+    MAST Observations query class.
+
+    Class for querying MAST observational data.
+    """
+
+    def __init__(self, mast_token=None):
+
+        super(ObservationsClass, self).__init__()
```

Comment:
Not something for now since it would be nice to keep consistency, at least within the mast module, but I wonder if we should start using the Python 3 style that I think would just be `super().__init__()`.

Reply:
Oh I didn't know that had changed. I will look into it and upgrade. I'm trying to Python 3-ify as I go along.